FunctionalTableData/TableSection.swift

```diff
@@ -164,6 +164,20 @@ extension Array where Element: TableSectionType {
 
 extension TableSection: Equatable {
 	public static func ==(lhs: TableSection, rhs: TableSection) -> Bool {
-		return lhs.key == rhs.key
+		return lhs.key == rhs.key &&
+			(lhs.header == nil && rhs.header == nil || lhs.header?.isEqual(rhs.header) ?? false) &&
+			(lhs.footer == nil && rhs.footer == nil || lhs.footer?.isEqual(rhs.footer) ?? false) &&
+			isEqual(lhs: lhs.rows, rhs: rhs.rows) &&
```

Comment:
The equality of the rows happen somewhere else, no need to do this here right?

Reply:
The `==` function is only used in 2 places in a closed source framework built on this framework. In order to ensure this equality function as defined remains correct, it needs to do row equality checks here. You are correct that in normal operation the diffing algorithm handles row comparisons separately outside of this equality function, but I think that might just be due to differing times that the diffing and this function were created.

Reply:
Wouldn't this essentially run them twice? Equality check on all rows when checking the TableSection equality and then equality check on all rows in `TableSectionChangeSet`?

Reply:
I ran the code and this `==` function isn't run during the diffing algorithm. I commented out this entire function and it was only used in 2 places, none of which were in the FunctionalTableData framework.

Reply:
😳